snf-ui/app/components/file-editor.js

```diff
@@ -0,0 +1,80 @@
+import {editFile} from 'snf-ui/snf/editor';
+import {showGroupsOnKeyUp} from 'snf-ui/views/application';
+import {unbindKeyboardShortcuts, bindKeyboardShortcuts} from 'snf-ui/snf/common';
+
+export default Ember.Component.extend({
+
+  didInsertElement: function() {
+    this._super();
+   
+    // remove the X button 
+    $('.close-reveal-modal').remove();
+ 
+    // disable shortcuts
+    unbindKeyboardShortcuts();
+ 
+    
+    // INITIALIZE EDITOR
+    var editor, filename, contents;
+    editor = this.$("#file-editor")[0];
+    filename = this.get('model.name');
+    var readOnly = this.get('readOnly');
+    
+    this.set('contentChanged', false);
+
+    if (!readOnly) {
+      $(".reveal-modal").attr('data-options', 'close_on_background_click:false;close_on_esc:false');
+    }
+
+    var onSave = () => { this.send('save'); }
+    var onChanged = (changed) => { this.set('contentChanged', changed); }
+
+    // reading file requires http request, thus editor  
+    this.set('isLoading', true);
+    this.get('model').read().then((contents) => {
+      this.set('isLoading', false);
+      editFile(editor, filename, contents, readOnly, onSave, onChanged);
+    });
+
+  },
+
+  actions: {
+    save() {
+        if (this.get('contentChanged')) {
+          let contents = editor.getValue();
+          var error500 = false;
+          this.sendAction('onSave', this.get('model'), contents, function(isTrue){
+            if (isTrue === false) {
+              error500 = true;
+            }
+          });
+          if (error500) {
```

Comment:
I am not sure if this will ever be handled since the callback:
```javascript
function(isTrue){
  if (isTrue === false) {
    error500 = true;
  }
}
```
will be executed once the file saving ends. Also the isTrue naming can be rather confusing. I would suggest the code should be moved inside the callback like:
```javascript
function(error){
	if(error) {
		$("#statusBar").text("An internal server error (500) occured.");
	} else {
		this.set('contentChanged', false);
		$("#statusBar").text("saved");
	}
}
```